integration_tests/cork_test.go

```diff
@@ -154,27 +150,134 @@ func (s *IntegrationTestSuite) TestCork() {
 			return true
 		}, 105*time.Second, 1*time.Second, "new vote period never seen")
 
-		s.T().Logf("checking for updated tick ranges in cellar")
+		s.T().Log("querying gravity module for logic call transaction")
+		aave_abi, err := AaveV2MetaData.GetAbi()
+		s.Require().NoError(err)
+
+		methodName := "claimAndUnstake"
+		method, err := aave_abi.Pack(methodName)
+		invalidationScope := crypto.Keccak256(method)
+		invalidationNonce := 1
+		s.Require().NoError(err)
+
+		kb, err := val.keyring()
+		s.Require().NoError(err)
+		clientCtx, err := s.chain.clientContext("tcp://localhost:26657", &kb, "val", val.keyInfo.GetAddress())
+		s.Require().NoError(err)
+
+		gravityQueryClient := gravityTypes.NewQueryClient(clientCtx)
+
 		s.Require().Eventuallyf(func() bool {
-			actualTickRange, err := s.getFirstTickRange()
-			if err != nil {
-				s.T().Logf("got error %e querying ticks", err)
-				return false
+			request := &gravityTypes.ContractCallTxRequest{
+				InvalidationNonce: uint64(invalidationNonce),
+				InvalidationScope: invalidationScope,
 			}
-			if expectedTickRange.Upper != actualTickRange.Upper {
-				s.T().Logf("wrong upper %s", actualTickRange.String())
-				return false
+			response, _ := gravityQueryClient.ContractCallTx(context.Background(), request)
+
+			if response != nil {
+				s.T().Logf("found logic call %v!", response.LogicCall)
+				return true
 			}
-			if expectedTickRange.Lower != actualTickRange.Lower {
-				s.T().Logf("wrong lower %s", actualTickRange.String())
-				return false
+
+			return false
+		}, 1*time.Minute, 5*time.Second, "cellar event never seen")
+
+		s.T().Logf("waiting for gravity to submit call to cellar")
+		s.Require().Eventuallyf(func() bool {
+			s.T().Log("querying gravity logic call events...")
+			ethClient, err := ethclient.Dial(fmt.Sprintf("http://%s", s.ethResource.GetHostPort("8545/tcp")))
+			s.Require().NoError(err)
+
+			// For non-anonymous events, the first log topic is a keccak256 hash of the
+			// event signature.
+			eventSignature := []byte("LogicCallEvent(bytes32,uint256,bytes,uint256)")
+			logicCallEventSignatureTopic := crypto.Keccak256Hash(eventSignature)
+
+			query := ethereum.FilterQuery{
+				FromBlock: nil,
+				ToBlock:   nil,
+				Addresses: []common.Address{
+					gravityContract,
+				},
+				Topics: [][]common.Hash{
+					{
+						logicCallEventSignatureTopic,
+					},
+				},
 			}
-			if expectedTickRange.Weight != actualTickRange.Weight {
-				s.T().Logf("wrong weight %s", actualTickRange.String())
-				return false
+			logs, err := ethClient.FilterLogs(context.Background(), query)
+			s.Require().NoError(err)
+			s.T().Logf("got %v gravity.submitLogicCall logs", len(logs))
+			ethClient.Close()
+
+			gravity_abi, err := GravityMetaData.GetAbi()
+			s.Require().NoError(err)
+
+			for i, log := range logs {
+				if len(log.Data) > 0 {
+					out, err := gravity_abi.Unpack("LogicCallEvent", log.Data)
```

Comment:
I think you can use `UnpackIntoInterface` and pass it a pointer to an empty struct.